src/or/entrynodes.c

```diff
@@ -1231,43 +1234,38 @@ entry_guard_is_listed,(guard_selection_t *gs, const entry_guard_t *guard))
   }
 }
 
-/**
- * Update the status of all sampled guards based on the arrival of a
- * new consensus networkstatus document.  This will include marking
- * some guards as listed or unlisted, and removing expired guards. */
-STATIC void
-sampled_guards_update_from_consensus(guard_selection_t *gs)
+/** Enumerate <b>sampled_entry_guards</b> smartlist in <b>gs</b>.
+ * For each <b>entry_guard_t</b> object in smartlist, do the following:
+ *  * Update <b>currently_listed</b> field to reflect if guard is listed
+ *    in guard selection <b>gs</b>.
+ *  * Set <b>unlisted_since_date</b> to approximate UNIX time of
+ *    unlisting if guard is unlisted (randomize within 20% of
+ *    REMOVE_UNLISTED_GUARDS_AFTER). Otherwise, set it to 0.
+ *
+ * Increment <b>n_changes</b> by a number of entries updated.
+ * Require <b>gs</b> and <b>n_changes</b> to be non-null pointers.
+ */
+static void
+sampled_guards_update_listedness(guard_selection_t *gs,
+                                 int *n_changes)
 {
   tor_assert(gs);
-  const int REMOVE_UNLISTED_GUARDS_AFTER =
-    (get_remove_unlisted_guards_after_days() * 86400);
-  const int unlisted_since_slop = REMOVE_UNLISTED_GUARDS_AFTER / 5;
-
-  // It's important to use only a live consensus here; we don't want to
-  // make changes based on anything expired or old.
-  if (live_consensus_is_missing(gs)) {
-    log_info(LD_GUARD, "Not updating the sample guard set; we have "
-             "no live consensus.");
-    return;
-  }
-  log_info(LD_GUARD, "Updating sampled guard status based on received "
-           "consensus.");
+  tor_assert(n_changes);
```

Comment:
Suggestion: Instead of taking an `n_changes` pointer argument, let's switch this function (and the other one) from `void` to `int`, and have it return the value of `n_changes`. I think that will be cleaner, since there won't be a need to do pointer operations.

Reply:
Did so (with `size_t`) in eea7c816335766b7bb15b07b060ae9df7cf20e2e.

Reply:
LGTM.